src/main/scala/dsptools/numbers/chisel_types/SIntTypeClass.scala

```diff
@@ -67,6 +72,13 @@ trait SIntSigned extends Any with Signed[SInt] with hasContext {
   }
   // isSignPositive, isSignNonZero, isSignNonPositive, isSignNonNegative derived from above (!)
   // abs requires ring (for overflow) so overridden later
+  def context_abs(a: SInt): SInt = {
+    context.overflowType match {
+      case Grow => Mux(a >= 0.S, a, 0.S -& a)
```

Comment:
@shunshou I wondered if I used context_- then I wouldn't need to do the match statement, because the context_- would do the right thing.  Is that right?  (with ShiftRegister, of course)